### ISSUE TITLE
Initial feedback

### ISSUE THREAD

**Issue by dgwyer** [ahmadawais/create-guten-block#26]
Just used this tool for the first time (great utility by the way!). Here's a couple of suggestions to make it more 'user friendly'.

* When installing `create-guten-block` for the first time it would be nice to run a series of questions for user input that will be used to create a config file (author, plugin URI, version). The idea is that every time a new plugin is created with `create-guten-block` these settings will be used to create the plugin header.

These questions could also be asked when creating a new plugin and if you just press enter then the value in the config file will be used. Also, some way of updating/editing the config file would be helpful.

* Make the plugin name be the name of the project when the `create-guten-block` command is executed. This would save time having to edit the plugin name.

i.e. Entering this `create-guten-block my-custom-block-plugin` would create a plugin with the following entry in the plugin header:

```php
/**
 * Plugin Name: My Custom Block Plugin
  ...
 */
```

i.e. Hyphens/underscore characters are removed and each word is then  capitalized.

**Comment by ahmadawais** [comment]
Hey, @dgwyer!
Thanks for trying out create-guten-block and giving these suggestions. I'll take that in for version 2.0.0 #11 I want this to stay as simple as possible and not so much configuration should be done by users.

**Comment by dgwyer** [comment]
Yes, it should not complicate things with unnecessary configuration.

I'd definitely consider adding in the name of the plugin name in the header though that would be useful.

And I'd recommend reducing the string instances of 'cgb' and 'CGB' as these will have to be replaced manually. I'd probably think about changing these string instances to very generic strings that wouldn't necessarily have to be changed immediately.

**Comment by helen** [comment]
I came to suggest the same things :) +1 to (skippable) prompts for customization. Doesn't need to be all of the plugin header fields IMO, but certainly the name, function/namespacing prefix vs. plugin directory name, author, and possibly choosing between GPLv2+ and MIT licenses (making that a first-class thought).

**Comment by ahmadawais** [comment]
OK, if anyone of you want to take a stab at it, more than welcomed.

The code for this exists in `init.js` file that acts as an initializer. I'm using pretty trivial `sed` replacements for the premade template code present in `cgb-scripts/template`.

https://github.com/ahmadawais/create-guten-block/blob/39c83b445921e506b07ae364fe14ac23809afe54/packages/cgb-scripts/scripts/init.js#L24-L85

Ideally, we can move towards using some sort of templating language here. 

I think `create-guten-block` should have sane-defaults. 
- Check if the name is valid as mentioned in #28 
- Then check if there's a parameter for custom name's something like `--names` we can use `commander` for that
https://github.com/ahmadawais/create-guten-block/blob/39c83b445921e506b07ae364fe14ac23809afe54/packages/create-guten-block/app/cli.js#L20-L37

- Use `inquirer` to get the dynamic names and search/replace them in the template like done above.

https://github.com/ahmadawais/create-guten-block/blob/39c83b445921e506b07ae364fe14ac23809afe54/packages/cgb-scripts/scripts/eject.js#L48-L54

An alternate approach would be to ask about the names by default and use the sane-defaults if there is `-y|--yes` argument just like `npm init -y` we could have `create-guten-block my-block -y`

PR's welcomed! 👍
